lib/jitter/sleep.go

```diff
@@ -15,11 +20,12 @@ func Jitter(baseMs, maxMs, attempts int) time.Duration {
 		return time.Duration(0)
 	}
 
-	// Check for overflows when computing base * 2 ** attempts.
-	if attemptsMaxMs := baseMs * (1 << attempts); attemptsMaxMs > 0 {
-		maxMs = min(maxMs, attemptsMaxMs)
+	// maxThreshold is used to make sure the multiplier does not cause any value overflow.
+	// The 1 million comes from `ms` and `ns` conversion.
+	maxThreshold := math.MaxInt / baseMs / 1_000_000
+	if multiplier := powerOfTwo(min(attempts, 20)); multiplier <= maxThreshold {
```

Comment:
Where does 20 come from?

Reply:
just a generous budget for 2^20

Reply:
Gotcha. It would be nice to add a docstring to `Jitter()` explaining the limitations since we're using it in `reader` and other projects.